var/spack/repos/builtin/packages/fluxbox/package.py

```diff
@@ -0,0 +1,49 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+#
+from spack import *
+
+
+class Fluxbox(AutotoolsPackage):
+    """Fluxbox is a windowmanager for X that was based on the Blackbox 0.61.1 code.
+ It is very light on resources and easy to handle but yet full of features
+ to make an easy, and extremely fast, desktop experience.
+ """
+
+    homepage = "http://fluxbox.org/"
+    url      = "http://sourceforge.net/projects/fluxbox/files/fluxbox/1.3.7/fluxbox-1.3.7.tar.gz"
+
+    version('1.3.7', 'd99d7710f9daf793e0246dae5304b595')
+
+    # depends_on('m4', type='build')
+    # depends_on('autoconf', type='build')
+    # depends_on('automake', type='build')
+    # depends_on('libtool', type='build')
```

Comment:
Can you either uncomment or remove commented dependencies?

Reply:
Sorry, I ' ve left them as I do not know which is the right policy:

I' ve tested it with dependency commented on all of our cluster deployment.

I' ve tried a build on a docker ubuntu and it seem to work.

Nevertheless I do not know which is the correct policy:
are automake and autoconf dependencies already ensured by deriving from AutotoolsPackage ?
Which is the suggested policy regarding base build tools like libtool, binutils, m4 ?

For me the easiest path is to remove all commented depends.

I' m willing to try one of the docker based approach for testing builds but not there yet.

Reply:
You need these dependencies only for packages that don't ship with a `configure` script. For instance if you have a `develop` version tied to a repository checkout.